packages/slate-react/src/components/editor.js

```diff
@@ -71,130 +72,64 @@ class Editor extends React.Component {
    * @param {Object} props
    */
 
-  constructor(props) {
-    super(props)
-    this.state = {}
-    this.tmp = {}
-    this.tmp.updates = 0
-    this.tmp.resolves = 0
-
-    // Resolve the plugins and create a stack and schema from them.
-    const plugins = this.resolvePlugins(props.plugins, props.schema)
-    const stack = Stack.create({ plugins })
-    const schema = Schema.create({ plugins })
-    this.state.schema = schema
-    this.state.stack = stack
-
-    // Run `onChange` on the passed-in value because we need to ensure that it
-    // is normalized, and queue the resulting change.
```

Comment:
Where has this logic gone? This is important to ensure that the editor never renders a `value` that is invalid according to the schema.

Reply:
@ianstormtaylor This part of logic has gone to `get value`

`get value` use the resolved stack, and use the resolved stack to get value after `onChange` function.

Reply:
Oh, I see.  Do you mean if stack changes and value is unchanged, we shall still run `onChange`?

Reply:
With the current logic, if the `<Editor>` is passed an invalid `value`, it will normalize it against the schema before rendering. And then after mounting it will ensure that the operations in that extra normalization pass are propagated to `onChange` so that the view layer is aware.